core/lib/Thelia/Controller/Admin/ProductController.php

```diff
@@ -1804,4 +1816,365 @@ protected function arrayHasEntries(array $data, array $entries)
+                    ->save();
+            } else {
+                $cpI18n = new ProductI18n();
+                $cpI18n
+                    ->setId($cpId)
+                    ->setLocale($opI18n->getLocale())
+                    ->setTitle($opI18n->getTitle())
+                    ->setChapo($opI18n->getChapo())
+                    ->setDescription($opI18n->getDescription())
+                    ->setPostscriptum($opI18n->getPostscriptum())
+                    ->setMetaTitle($opI18n->getMetaTitle())
+                    ->setMetaDescription($opI18n->getMetadescription())
+                    ->setMetaKeywords($opI18n->getMetaKeywords())
+                    ->save();
+            }
+        }
+    }
+
+    public function setCloneImage($opId, $cpId)
+    {
+        // Get original product images
+        $opImages = ProductImageQuery::create()
+            ->findByProductId($opId);
+
+        // Set clone's images
+        foreach ($opImages as $opImage) {
+            $cpImage = new ProductImage();
+            $cpImage
+                ->setProductId($cpId)
+                ->setFile($opImage->getFile())
+                ->setVisible($opImage->getVisible())
+                ->setPosition($opImage->getPosition())
+                ->save();
+
+            // Get original product images I18n
+            $opImageI18ns = ProductImageI18nQuery::create()
+                ->findById($opImage->getId());
+
+            // Set clone's images I18n
+            foreach ($opImageI18ns as $opImageI18n) {
+                $cpImageI18n = new ProductImageI18n();
+                $cpImageI18n
+                    ->setId($cpImage->getId())
+                    ->setLocale($opImageI18n->getLocale())
+                    ->setTitle($opImageI18n->getTitle())
+                    ->setChapo($opImageI18n->getChapo())
+                    ->setDescription($opImageI18n->getDescription())
+                    ->setPostscriptum($opImageI18n->getPostscriptum())
+                    ->save();
+            }
+        }
+    }
+
+    public function setCloneDocument($opId, $cpId)
+    {
+        // Get original product documents
+        $opDocuments = ProductDocumentQuery::create()
+            ->findByProductId($opId);
+
+        // Set clone's documents
+        foreach ($opDocuments as $opDocument) {
+            $cpDocument = new ProductDocument();
+            $cpDocument
+                ->setProductId($cpId)
+                ->setFile($opDocument->getFile())
+                ->setVisible($opDocument->getVisible())
+                ->setPosition($opDocument->getPosition())
+                ->save();
+
+            // Get original product documents I18n
+            $opDocumentI18ns = ProductDocumentI18nQuery::create()
+                ->findById($opDocument->getId());
+
+            // Set clone's documents I18n
+            foreach ($opDocumentI18ns as $opDocumentI18n) {
+                $cpDocumentI18n = new ProductDocumentI18n();
```

Comment:
Have a look at Propel's copy() method, it may saves you from writing tedious code.